utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -104,14 +105,16 @@ fn maybe_filter_value<T: TrieValue>(value: T, trie_null_value: T, null_value: T)
 /// For more information:
 /// - [ICU Site design doc](http://site.icu-project.org/design/struct/utrie)
 /// - [ICU User Guide section on Properties lookup](https://unicode-org.github.io/icu/userguide/strings/properties.html#lookup)
-#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+// serde impls in crate::serde
 #[derive(Debug, Eq, PartialEq, Yokeable, ZeroFrom)]
 pub struct CodePointTrie<'trie, T: TrieValue> {
-    header: CodePointTrieHeader,
-    #[cfg_attr(feature = "serde", serde(borrow))]
-    index: ZeroVec<'trie, u16>,
-    #[cfg_attr(feature = "serde", serde(borrow))]
-    data: ZeroVec<'trie, T>,
+    pub(crate) header: CodePointTrieHeader,
+    pub(crate) index: ZeroVec<'trie, u16>,
+    pub(crate) data: ZeroVec<'trie, T>,
```

Comment:
Question: Are we trimming the last value off the ZeroVec, or are we leaving it there? Seems like we're leaving the value in place but never reading from it except in the constructor?